examples/stats/exporter/prometheus.js

```diff
@@ -14,4 +14,172 @@
+  "repl/latency",
+  MeasureUnit.MS,
+  "The latency in milliseconds per REPL loop"
+);
+
+// Counts/groups the lengths of lines read in.
+const mLineLengths = stats.createMeasureInt64(
+  "repl/line_lengths",
+  MeasureUnit.BYTE,
+  "The distribution of line lengths"
+);
+
+// Create a stream to read our file
+const stream = fs.createReadStream("./test.txt");
+
+// Create an interface to read and process our file line by line
+const lineReader = readline.createInterface({ input: stream });
+
+const tagKeys = ["method", "status"];
+
+// Create & Register the view.
+/*const latencyView = */stats.createView(
+  "demo/latency",
+  mLatencyMs,
+  AggregationType.DISTRIBUTION,
+  tagKeys,
+  "The distribution of the repl latencies",
+  // Latency in buckets:
+  // [>=0ms, >=25ms, >=50ms, >=75ms, >=100ms, >=200ms, >=400ms, >=600ms, >=800ms, >=1s, >=2s, >=4s, >=6s]
+  [0, 25, 50, 75, 100, 200, 400, 600, 800, 1000, 2000, 4000, 6000]
+);
+//stats.registerView(latencyView);
+
+// Create & Register the view.
+/*const lineCountView = */stats.createView(
+  "demo/lines_in",
+  mLineLengths,
+  AggregationType.COUNT,
+  tagKeys,
+  "The number of lines from standard input"
+);
+//stats.registerView(lineCountView);
+
+// Create & Register the view.
+/*const lineLengthView = */stats.createView(
+  "demo/line_lengths",
+  mLineLengths,
+  AggregationType.DISTRIBUTION,
+  tagKeys,
+  "Groups the lengths of keys in buckets",
+  // Bucket Boudaries:
+  // [>=0B, >=5B, >=10B, >=15B, >=20B, >=40B, >=60B, >=80, >=100B, >=200B, >=400, >=600, >=800, >=1000]
+  [0, 5, 10, 15, 20, 40, 60, 80, 100, 200, 400, 600, 800, 1000]
+);
+//stats.registerView(lineLengthView);
+
+// The begining of our REPL loop
+let [_, startNanoseconds] = process.hrtime();
+let endNanoseconds;
+
+// REPL is the read, evaluate, print and loop
+lineReader.on("line", function(line) {
+  // Read
+  try {
+    const processedLine = processLine(line); // Evaluate
+    console.log(processedLine); // Print
+
+    // Registers the end of our REPL
+    [_, endNanoseconds] = process.hrtime();
+
+    const tags = { method: "repl", status: "OK" };
+
+    stats.record({
+      measure: mLineLengths,
+      tags,
+      value: processedLine.length
+    });
+
+    stats.record({
+      measure: mLatencyMs,
+      tags,
+      value: sinceInMilliseconds(endNanoseconds, startNanoseconds)
+    });
+  } catch (err) {
+    const errTags = { method: "repl", status: "ERROR", error: err.message };
+    stats.record({
+      measure: mLatencyMs,
+      errTags,
+      value: sinceInMilliseconds(endNanoseconds, startNanoseconds)
+    });
+  }
+
+  // Restarts the start time for the REPL
+  startNanoseconds = endNanoseconds;
+});
+
+/**
+ * The default export interval is 60 seconds.
+ * The thread with must live for at least the interval past any
+ * metrics that must be collected, or some risk being lost if they are recorded
+ * after the last export.
+ */
+setTimeout(function() {
```

Comment:
I think we should remove this block. This is not applicable for Prometheus exporter.

Reply:
Fixed: 643941cdf7d80e5064e00bdf23cbccae117a519f